src/main/java/com/github/wvengen/maven/proguard/ProGuardMojo.java

```diff
@@ -48,132 +48,118 @@
 import org.apache.maven.plugin.MojoExecutionException;
 import org.apache.maven.plugin.MojoFailureException;
 import org.apache.maven.plugin.logging.Log;
+import org.apache.maven.plugins.annotations.Component;
+import org.apache.maven.plugins.annotations.LifecyclePhase;
+import org.apache.maven.plugins.annotations.Mojo;
+import org.apache.maven.plugins.annotations.Parameter;
+import org.apache.maven.plugins.annotations.ResolutionScope;
 import org.apache.maven.project.MavenProject;
 import org.apache.maven.project.MavenProjectHelper;
 import org.apache.tools.ant.DefaultLogger;
 import org.apache.tools.ant.Project;
 import org.apache.tools.ant.taskdefs.Java;
+import org.codehaus.plexus.archiver.Archiver;
 import org.codehaus.plexus.archiver.jar.JarArchiver;
 
 /**
  * Runs ProGuard as part of the build.
- *
- * @goal proguard
- * @phase package
- * @description Create small jar files using ProGuard
- * @requiresDependencyResolution compile
- * @threadSafe
  */
-
+@Mojo(
+	name = "proguard",
+	defaultPhase = LifecyclePhase.PACKAGE,
+	requiresDependencyResolution = ResolutionScope.COMPILE,
+	threadSafe = true
+)
 public class ProGuardMojo extends AbstractMojo {
 
 	/**
 	 * Set this to 'true' to bypass ProGuard processing entirely.
-	 *
-	 * @parameter property="proguard.skip"
 	 */
+	@Parameter(property = "proguard.skip")
 	private boolean skip;
 
 	/**
 	 * Recursively reads configuration options from the given file.
-	 *
-	 * @parameter default-value="${basedir}/proguard.conf"
 	 */
+	@Parameter(defaultValue = "${project.basedir}/proguard.conf")
```

Comment:
The variant without the `project.` prefix is deprecated. See https://maven.apache.org/guides/introduction/introduction-to-the-pom.html#Project_Model_Variables
> or the prefix omitted entirely - these forms are now deprecated and should not be used